core/variant/variant_call.cpp

```diff
@@ -753,40 +753,56 @@ struct _VariantCall {
 	static PackedInt32Array func_PackedByteArray_decode_s32_array(PackedByteArray *p_instance) {
 		uint64_t size = p_instance->size();
 		PackedInt32Array dest;
-		ERR_FAIL_COND_V_MSG(size < sizeof(int32_t), dest, "Size didn't match array of size int32_t, maybe you are trying to convert to the wrong type?");
+		if (size == 0) {
+			return dest;
+		}
+		static_assert(sizeof(dest[0]) == sizeof(int32_t));
+		ERR_FAIL_COND_V_MSG(size % sizeof(int32_t), dest, "Size didn't match array of size int32_t, maybe you are trying to convert to the wrong type?");
```

Comment:
>Size didn't match array of size int32_t

Shouldn't it be something like _"Size didn't match size of array of int32_t"_ instead? :thinking:

Reply:
This was just the same text as before. But I changed it to be more precise:
`"PackedByteArray size not divisible by sizeof int32_t, maybe you are trying to convert to the wrong type?"`